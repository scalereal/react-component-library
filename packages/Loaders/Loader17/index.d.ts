import * as React from 'react';
import { WithStyle } from '../../utils';
import { LoaderProps } from './types';
declare const Loader17: React.FC<LoaderProps> & WithStyle;
export default Loader17;
