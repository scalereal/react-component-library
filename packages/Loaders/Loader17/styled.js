var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
var mixin = function (size) {
    switch (size) {
        case "XXS":
            return "\n                height: 10px;\n                width: 10px;\n                border: 1px solid transparent;\n                &:before{\n                    top: 2px;\n                    right: 2px;\n                    bottom: 2px;\n                    left: 2px;\n                    border: 1px solid transparent;\n                }\n           ";
        case "XS":
            return "\n                height: 20px;\n                width: 20px;\n                border: 2px solid transparent;\n                &:before{\n                    top: 5px;\n                    right: 5px;\n                    bottom: 5px;\n                    left: 5px;\n                    border: 2px solid transparent;\n                }\n            ";
        case "S":
            return "\n            height: 40px;\n            width: 40px;\n            border: 4px solid transparent;\n            &:before{\n                top: 8px;\n                right: 8px;\n                bottom: 8px;\n                left: 8px;\n                border: 4px solid transparent;\n            }\n            ";
        case "M":
            return "\n            height: 60px;\n            width: 60px;\n            border: 5px solid transparent;\n            &:before{\n                top: 10px;\n                right: 10px;\n                bottom: 10px;\n                left: 10px;\n                border: 5px solid transparent;\n            }\n            ";
        case "L":
            return "\n                border: 6px solid transparent;\n                height: 80px;\n                width: 80px;\n            &:before{\n                top: 10px;\n                right: 10px;\n                bottom: 10px;\n                left: 10px;\n                border: 5px solid transparent;\n            }\n            ";
        case "XL":
            return "\n            height: 120px;\n            width: 120px;\n            border: 6px solid transparent;\n            &:before{\n                top: 20px;\n                right: 20px;\n                bottom: 20px;\n                left: 20px;\n                border: 6px solid transparent;\n            }\n            ";
        default:
            return "\n        height: 115px;\n        width: 115px;\n        border: 5px solid transparent;\n        &:before{\n            top: 20px;\n            right: 20px;\n            bottom: 20px;\n            left: 20px;\n            border: 5px solid transparent;\n        }\n       ";
    }
};
export var Loader = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    ", "\n\n    border-top-color: ", ";\n    border-bottom-color: ", ";\n    border-radius: 50%;\n    position: relative;\n    -webkit-animation: spin 3s linear infinite;\n            animation: spin 3s linear infinite;\n\n    &:before{\n        content: \"\";\n        position: absolute;\n        border-top-color: ", ";\n        border-bottom-color: ", ";\n        border-radius: 50%;\n        -webkit-animation: spin 1.5s linear infinite;\n        animation: spin 1.5s linear infinite;\n\n    }\n\n    @-webkit-keyframes spin {\n        from {\n            -webkit-transform: rotate(0deg);\n            transform: rotate(0deg);\n        }\n        to {\n            -webkit-transform: rotate(360deg);\n            transform: rotate(360deg);\n        }\n    }\n\n    @keyframes spin {\n        from {\n            -webkit-transform: rotate(0deg);\n            transform: rotate(0deg);\n        }\n        to {\n            -webkit-transform: rotate(360deg);\n            transform: rotate(360deg);\n        }\n    }\n\n"], ["\n    ", "\n\n    border-top-color: ", ";\n    border-bottom-color: ", ";\n    border-radius: 50%;\n    position: relative;\n    -webkit-animation: spin 3s linear infinite;\n            animation: spin 3s linear infinite;\n\n    &:before{\n        content: \"\";\n        position: absolute;\n        border-top-color: ", ";\n        border-bottom-color: ", ";\n        border-radius: 50%;\n        -webkit-animation: spin 1.5s linear infinite;\n        animation: spin 1.5s linear infinite;\n\n    }\n\n    @-webkit-keyframes spin {\n        from {\n            -webkit-transform: rotate(0deg);\n            transform: rotate(0deg);\n        }\n        to {\n            -webkit-transform: rotate(360deg);\n            transform: rotate(360deg);\n        }\n    }\n\n    @keyframes spin {\n        from {\n            -webkit-transform: rotate(0deg);\n            transform: rotate(0deg);\n        }\n        to {\n            -webkit-transform: rotate(360deg);\n            transform: rotate(360deg);\n        }\n    }\n\n"])), function (pr) { return pr.size && mixin(pr.size); }, function (pr) { return pr.Color ? pr.Color : '#9C27B0'; }, function (pr) { return pr.Color ? pr.Color : '#9C27B0'; }, function (pr) { return pr.innerColor ? pr.innerColor : '#BA68C8'; }, function (pr) { return pr.innerColor ? pr.innerColor : '#BA68C8'; });
export var LoaderWrapper = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    ", "\n    display: ", ";\n"], ["\n    ", "\n    display: ", ";\n"])), function (pr) { return pr.disabled === true ? 'pointer-events: none;opacity: 0.4;' : ''; }, function (pr) { return pr.visible ? 'inline-block' : 'none'; });
export var StyledWrapper = styled.div(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n    -webkit-box-flex: 0;\n    -ms-flex: 0 0 25%;\n    flex: 0 0 25%;\n    border: 1px solid rgba(255, 255, 255, 0.1);\n    -webkit-box-sizing: border-box;\n    box-sizing: border-box;\n    margin: 0;\n    position: relative;\n    display: -webkit-box;\n    display: -ms-flexbox;\n    display: flex;\n    -webkit-box-pack: center;\n    -ms-flex-pack: center;\n    justify-content: center;\n    -webkit-box-align: center;\n    -ms-flex-align: center;\n    align-items: center;\n    overflow: hidden;\n"], ["\n    -webkit-box-flex: 0;\n    -ms-flex: 0 0 25%;\n    flex: 0 0 25%;\n    border: 1px solid rgba(255, 255, 255, 0.1);\n    -webkit-box-sizing: border-box;\n    box-sizing: border-box;\n    margin: 0;\n    position: relative;\n    display: -webkit-box;\n    display: -ms-flexbox;\n    display: flex;\n    -webkit-box-pack: center;\n    -ms-flex-pack: center;\n    justify-content: center;\n    -webkit-box-align: center;\n    -ms-flex-align: center;\n    align-items: center;\n    overflow: hidden;\n"])));
var templateObject_1, templateObject_2, templateObject_3;
