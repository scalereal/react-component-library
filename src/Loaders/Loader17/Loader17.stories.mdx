import Loader17 from '.';
import { SizeMap } from '../types';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Loaders/Loader17"
    component={Loader17}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        innerColor: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <Loader17 {...args} />
    </div>
);

<Preview>
    <Story name="Loader17" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexWrap: 'wrap' }}>
        <Loader17 size="S" Color="green" />
        <Loader17 size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexDirection: 'column' }}>
        <Loader17 size="XXS" />
        <Loader17 size="XS" />
        <Loader17 size="S" />
        <Loader17 size="M" />
        <Loader17 size="L" />
        <Loader17 size="XL" />
    </div>
</Preview>

### Props

<Props story="Loader17" />
