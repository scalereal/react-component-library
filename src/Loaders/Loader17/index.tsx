import React from 'react';
import { WithStyle } from '../../utils';
import { Loader, LoaderWrapper, StyledWrapper } from './styled';
import { LoaderProps } from './types';

const Loader17:React.FC<LoaderProps> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <StyledWrapper {...props}>
                <Loader {...props}/>
            </StyledWrapper>
        </LoaderWrapper>
    );
    
}));

Loader17.displayName="Loader17";
Loader17.defaultProps = {
    size:'S',
    display: true
};

export default Loader17;
