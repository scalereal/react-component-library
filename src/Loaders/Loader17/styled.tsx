import styled from 'styled-components';
import { LoaderProps } from './types';

const mixin = (size:string) => {
    switch (size) {
        case "XXS":
            return `
                height: 10px;
                width: 10px;
                border: 1px solid transparent;
                &:before{
                    top: 2px;
                    right: 2px;
                    bottom: 2px;
                    left: 2px;
                    border: 1px solid transparent;
                }
           `;
        case "XS":
            return `
                height: 20px;
                width: 20px;
                border: 2px solid transparent;
                &:before{
                    top: 5px;
                    right: 5px;
                    bottom: 5px;
                    left: 5px;
                    border: 2px solid transparent;
                }
            `;
        case "S":
            return `
            height: 40px;
            width: 40px;
            border: 4px solid transparent;
            &:before{
                top: 8px;
                right: 8px;
                bottom: 8px;
                left: 8px;
                border: 4px solid transparent;
            }
            `;
        
        case "M":
            return `
            height: 60px;
            width: 60px;
            border: 5px solid transparent;
            &:before{
                top: 10px;
                right: 10px;
                bottom: 10px;
                left: 10px;
                border: 5px solid transparent;
            }
            `;

        case "L":
            return `
                border: 6px solid transparent;
                height: 80px;
                width: 80px;
            &:before{
                top: 10px;
                right: 10px;
                bottom: 10px;
                left: 10px;
                border: 5px solid transparent;
            }
            `;
        
        case "XL":
            return `
            height: 120px;
            width: 120px;
            border: 6px solid transparent;
            &:before{
                top: 20px;
                right: 20px;
                bottom: 20px;
                left: 20px;
                border: 6px solid transparent;
            }
            `;
  
      default:
        return `
        height: 115px;
        width: 115px;
        border: 5px solid transparent;
        &:before{
            top: 20px;
            right: 20px;
            bottom: 20px;
            left: 20px;
            border: 5px solid transparent;
        }
       `;
     }
 };

export const Loader = styled.div<LoaderProps>`
    ${pr=> pr.size && mixin(pr.size)}

    border-top-color: ${pr=>pr.Color?pr.Color:'#9C27B0'};
    border-bottom-color: ${pr=>pr.Color?pr.Color:'#9C27B0'};
    border-radius: 50%;
    position: relative;
    -webkit-animation: spin 3s linear infinite;
            animation: spin 3s linear infinite;

    &:before{
        content: "";
        position: absolute;
        border-top-color: ${pr=>pr.innerColor?pr.innerColor:'#BA68C8'};
        border-bottom-color: ${pr=>pr.innerColor?pr.innerColor:'#BA68C8'};
        border-radius: 50%;
        -webkit-animation: spin 1.5s linear infinite;
        animation: spin 1.5s linear infinite;

    }

    @-webkit-keyframes spin {
        from {
            -webkit-transform: rotate(0deg);
            transform: rotate(0deg);
        }
        to {
            -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
        }
    }

    @keyframes spin {
        from {
            -webkit-transform: rotate(0deg);
            transform: rotate(0deg);
        }
        to {
            -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
        }
    }

`;

export const LoaderWrapper = styled.div<LoaderProps>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    display: ${pr=>pr.display?'inline-block':'none'};
`;

export const StyledWrapper = styled.div<LoaderProps>`
    -webkit-box-flex: 0;
    -ms-flex: 0 0 25%;
    flex: 0 0 25%;
    border: 1px solid rgba(255, 255, 255, 0.1);
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    margin: 0;
    position: relative;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    overflow: hidden;
`;
