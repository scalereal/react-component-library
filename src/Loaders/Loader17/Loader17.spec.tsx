import { render } from '@testing-library/react';
import * as React from 'react';
import Loader17 from '.';

describe('Loader17 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<Loader17/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <Loader17
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <Loader17 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});