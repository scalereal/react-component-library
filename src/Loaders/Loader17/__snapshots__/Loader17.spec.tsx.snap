// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader17 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz hdkynM"
  >
    <div
      class="sc-dlfnbm bIJBxZ"
    >
      <div
        class="sc-bdfBwQ iRLYaI"
      />
    </div>
  </div>
</div>
`;

exports[`Loader17 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz hdkynM"
  >
    <div
      class="sc-dlfnbm bIJBxZ"
    >
      <div
        class="sc-bdfBwQ klSzIP"
      />
    </div>
  </div>
</div>
`;

exports[`Loader17 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz hdkynM"
  >
    <div
      class="sc-dlfnbm bIJBxZ"
    >
      <div
        class="sc-bdfBwQ huCtiX"
      />
    </div>
  </div>
</div>
`;
